app/services/application_search.rb

```diff
@@ -0,0 +1,57 @@
+class ApplicationSearch
+
+  attr_reader :error_message
+
+  def initialize(reference, current_user)
```

Comment:
I would suggest looking at the option of not passing `current_user` here, but the `ApplicationPolicy` instead. Because bellow you're testing if the office matches, but this condition can change by the policy and this could bypass it.

If you want I can have a look at how to use Pundit this way to help..